### PR TITLE
User/jml/remove fms2io legacy f90

### PR DESCRIPTION
**Pull request by wrongkindofdoctor**
**Description**
Removed the unnecessary legacy.F90 file. The functionality is moved to axis_edges in axis_utils2.F90 in #286, so this PR should be merged in after #286.
**How Has This Been Tested?**
Travis CI
**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,3 +26,8 @@ sequential patch number (starting from `01`).
 - Fixed `time_interp_missing` parameter in [time_interp/time_interp_external.F90](time_interp/time_interp_external.F90) to be within range when compiled in mixed-mode.
 - reverted `QUAD_PRECISION` cpp macro behavior to pre-Xanadu behavior in [include/fms_platform.h](include/fms_platform.h)
 - Fixed a GNU compiler issue with the logical check to set the netCDF fill value in `mpp_io_write` by separating the logical `.AND.` into nested `if` statements.
+## [2020.01] - 2020-##-##
+### Added
+### Fixed
+### Removed
+- `fms2_io/legacy.F90` The `axis_edges` function was moved to [axis_utils/axis_utils2.F90](axis_utils/axis_utils2.F90).
```
Comment (thomas-robinson) [inline CHANGELOG.md]: Do we need this?  The file never existed in any previous release.

Comment (wrongkindofdoctor) [reply CHANGELOG.md]: Good point. I'll remove it.

**Comment by uramirez8707** [comment]
@wrongkindofdoctor You need to change the order in https://github.com/NOAA-GFDL/FMS/blob/7788b781ac4e7fecf0d6cebc239206242305ac36/Makefile.am#L9, so that axis_utils can compile before diag_manager

**Comment by wrongkindofdoctor** [comment]
@uramirez8707 OMG, THANK YOU!!!! I was just going to send out a call for help because I could not figure out what I was missing.

**Comment by uramirez8707** [comment]
@wrongkindofdoctor getting rid of this line should fix it?
https://github.com/NOAA-GFDL/FMS/blob/7788b781ac4e7fecf0d6cebc239206242305ac36/libFMS/Makefile.am#L37

**Pull request merged**